ethcore/private-tx/src/log.rs

```diff
@@ -0,0 +1,412 @@
+// Copyright 2015-2018 Parity Technologies (UK) Ltd.
+// This file is part of Parity.
+
+// Parity is free software: you can redistribute it and/or modify
+// it under the terms of the GNU General Public License as published by
+// the Free Software Foundation, either version 3 of the License, or
+// (at your option) any later version.
+
+// Parity is distributed in the hope that it will be useful,
+// but WITHOUT ANY WARRANTY; without even the implied warranty of
+// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+// GNU General Public License for more details.
+
+// You should have received a copy of the GNU General Public License
+// along with Parity.  If not, see <http://www.gnu.org/licenses/>.
+
+//! Private transactions logs.
+
+use ethereum_types::{H256, Address};
+use std::collections::{HashMap};
+use std::fs::{File};
+use std::path::{PathBuf};
+use std::sync::{Arc};
+use std::time::{SystemTime, UNIX_EPOCH};
+use parking_lot::{RwLock};
+use serde::ser::{Serializer, SerializeSeq};
+use error::{Error};
+
+/// Maximum amount of stored private transaction logs.
+const MAX_JOURNAL_LEN: usize = 1000;
+
+/// Maximum period for storing private transaction logs.
+/// Older logs will not be processed, 20 days
+const MAX_STORING_TIME: u64 = 60 * 60 * 24 * 20;
+
+/// Current status of the private transaction
+#[derive(Clone, Serialize, Deserialize, Debug, PartialEq)]
+pub enum PrivateTxStatus {
+	/// Private tx was created but no validation received yet
+	Created,
+	/// Several validators (but not all) validated the transaction
+	Validating,
+	/// All validators validated the private tx
+	/// Corresponding public tx was created and added into the pool
+	Deployed,
+}
+
+/// Information about private tx validation
+#[derive(Clone, Serialize, Deserialize)]
+pub struct ValidatorLog {
+	/// Account of the validator
+	pub account: Address,
+	/// Validation flag
+	pub validated: bool,
```

Comment:
Why are both `validated` and `validation_timestamp` needed? 

AFAIU, you should not have a `validation_timestamp` unless validated it true?

Reply:
Timestamp removed